clang/test/CodeGen/integer-wrap.c

```diff
@@ -0,0 +1,77 @@
+// Check that -fsanitize=signed-integer-wrap instruments with -fwrapv
+// RUN: %clang_cc1 -fwrapv -triple x86_64-apple-darwin -emit-llvm -o - %s -fsanitize=signed-integer-wrap | FileCheck %s --check-prefix=CHECKSIW
+
+// Check that -fsanitize=signed-integer-overflow doesn't instrument with -fwrapv
+// RUN: %clang_cc1 -fwrapv -triple x86_64-apple-darwin -emit-llvm -o - %s -fsanitize=signed-integer-overflow | FileCheck %s --check-prefix=CHECKSIO
```

Comment:
Wait a second, why are you testing `signed-integer-overflow`? This is already tested in clang/test/codegen/integer-overflow.c. I didn't think your patch is changing the behavior of sio, only adding support for siw?

Reply:
My patch does not touch SIO but in my testing I wanted it to be clear that SIO and SIW behave differently when `-fwrapv` is enabled. The tests more or less show this difference .

Reply:
I moved the tests in [5497e8b](https://github.com/llvm/llvm-project/pull/80089/commits/5497e8bc6849bf64c1158ff16b4aa04fd9141920); as such, I am no longer adding tests related to SIO.